docs/skill-guide.md

```diff
@@ -0,0 +1,230 @@
+            logger.info("[{}]: found number of agents={}".format(self.context.agent_name, nb_agents_found))
+            self.received_search_count += 1
+
+    def teardown(self) -> None:
+        """
+        Teardown the handler.
+
+        :return: None
+        """
+        pass
+```
+
+We create a handler which is registered for the `oef` protocol. Whenever it receives a search result, we log the number of agents returned in the search - the agents matching the search query - and update the counter of received searches.
+
+Note, how the handler simply reacts to incoming events (i.e. messages). It could initiate further actions, however, they are still reactions to the upstream search event.
+
+
+## Step 4: Develop a Task
+
+We have implemented a behaviour and a handler. We conclude by implementing a task. Here we can implement background logic. We will implement a trivial check on the difference between the amount of search requests sent and responses received.
+
+
+```python
+import logging
+
+from aea.skills.base import Task
+
+logger = logging.getLogger("aea.my_search_skill")
+
+
+class MySearchTask(Task):
+    """This class scaffolds a task."""
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def execute(self) -> None:
+        """
+        Implement the task execution.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        my_search_behaviour = self.context.behaviours[0]
+        my_search_task = self.context.tasks[0]
+        my_search_behaviour
+        logger.info("[{}]: number of search requests sent={} vs. number of search responses received={}".format(self.context.agent_name,
+        	                    my_search_behaviour.sent_search_count,
+        	                    my_search_task.received_search_count)
+        )
+
+    def teardown(self) -> None:
+        """
+        Implement the task teardown.
+
+        :return: None
+        """
+	    pass
+```
+
+Note, how we have access to other objects in the skill via `self.context`.
+
+
+## Step 5: Create the config file
+
+Based on our skill components above, we create the following config file:
+
+```yaml
+name: my_search
+authors: Fetch.AI Limited
+version: 0.1.0
+license: Apache 2.0
+url: ""
+description: 'A simple search skill utilising the OEF.'
+behaviours:
+  - behaviour:
+      class_name: MySearchBehaviour
+      args: {}
+handlers:
+  - handler:
+      class_name: MySearchHandler
+      args: {}
+tasks:
+  - task:
+      class_name: MySearchTask
+      args: {}
+shared_classes: []
+protocols: ["oef"]
+dependencies: []
+```
+
+## Step 6: Run the agent
+
+We first start an oef node (see the <a href="../connection/">connection section</a> for more details) in a separate terminal window:
+
+```bash
+python scripts/oef/launch.py -c ./scripts/oef/launch_config.json
+```
+
+We can then launch our agent:
+```bash
+aea run
```

Comment:
The default logging level should be `info`, so all the log messages in the skill code should work. But worth double-check. Or, explicitly set the logging level with `-v INFO` or `-v info`